### PR TITLE
UD-5954 set default text

### PR DESCRIPTION
**Pull request by SashaBayan**
This PR does the DOM manipulation to set the text of the input as we described in our meeting.

https://youtu.be/xl5XLqkgft0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
   "peerDependencies": {
     "autoprefixer": "^9.0.2",
     "cssnano": "^4.0.5",
-    "daterangepicker": "^3.0.3",
+    "daterangepicker": "git://github.com/UrbanDoor/daterangepicker.git",
```
Comment (ianwremmel) [inline package.json]: Why this change? What changes are in there?

File: src/components/daterangepicker/daterangepicker.vue [hunk 2]
```diff
@@ -108,10 +119,18 @@ export default {
      * @param {moment.Moment} end
      */
     emit(start, end) {
+      if (start.isSame(end, 'day')) {
+        setTimeout(setDefaultText, 0);
```
Comment (ianwremmel) [inline src/components/daterangepicker/daterangepicker.vue]: Why does this need to be done in a `setTimeout`? would `this.$nextTick` be more appropriate?

File: src/components/daterangepicker/daterangepicker.vue [hunk 3]
```diff
@@ -13,6 +13,11 @@ import {get} from 'lodash';
 import {locale, toDateRangePickerValue, toLocaleString} from '../../lib/date';
 import {attributeToClassSelector} from '../../lib/css-selector';
 
+function setDefaultText() {
+  document.getElementsByClassName('two fields')[0].value =
+    'Arrive   →   Depart';
```
Comment (ianwremmel) [inline src/components/daterangepicker/daterangepicker.vue]: Can we use the values from the locale object instead of hardcoding them? That should make `i18n` easier later.

File: src/components/daterangepicker/daterangepicker.vue [hunk 4]
```diff
@@ -75,20 +80,26 @@ export default {
...
         minDate: toDateRangePickerValue(new Date()),
         parentEl: this.parentEl && attributeToClassSelector(this.parentEl),
+        onOutsideClick: this.clearDate,
         locale
       },
       (start, end) => {
         this.emit(start, end);
       }
     );
+    if (this.start === this.end) {
```
Comment (ianwremmel) [inline src/components/daterangepicker/daterangepicker.vue]: do we need this logic? Isn't checking for falsiness enough?

File: src/components/daterangepicker/daterangepicker.vue [hunk 5]
```diff
@@ -75,20 +80,26 @@ export default {
...
+      setTimeout(setDefaultText, 0);
+    }
+    if (this.start === undefined || this.end === undefined) {
```
Comment (ianwremmel) [inline src/components/daterangepicker/daterangepicker.vue]: Can these be checked for falsy rather than explicitly undefined?

File: package.json (hunk 1 above)
Comment (SashaBayan) [reply package.json]: I added an `onOutsideClick` event handler to our fork of DateRangePicker. Without this, the DRP will update itself with the date values (if none selected by user, start and end dates will both be set to the current day) and overwrite the `Arrive / Depart` text.

File: src/components/daterangepicker/daterangepicker.vue (hunk 4 above)
Comment (SashaBayan) [reply src/components/daterangepicker/daterangepicker.vue]: Checking date equivalence is our stop-gap for detecting whether a user has entered dates or not. The DRP will update its values via its internal lifecycle, which will percolate up to our Vue wrapper and set `this.start` and `this.end`, which means falsiness is not the only case we need to check against.

File: package.json (hunk 1 above)
Comment (SashaBayan) [reply package.json]: https://github.com/UrbanDoor/daterangepicker/commit/da32ba995e424072f5ca8be669d5b3deabb65012

File: src/components/daterangepicker/daterangepicker.vue (hunk 3 above)
Comment (SashaBayan) [reply src/components/daterangepicker/daterangepicker.vue]: As discussed, we will take care of this once we have a more set strategy for how to handle `il8n` in Press

File: src/components/daterangepicker/daterangepicker.vue [hunk 6]
```diff
@@ -13,6 +13,11 @@ import {get} from 'lodash';
 import {locale, toDateRangePickerValue, toLocaleString} from '../../lib/date';
 import {attributeToClassSelector} from '../../lib/css-selector';
 
+function setDefaultText() {
+  document.getElementsByClassName('two fields')[0].value =
```
Comment (shentao) [inline src/components/daterangepicker/daterangepicker.vue]: Don’t you think that relying on class names for selecting elements might be prone to bugs?

Comment (SashaBayan) [reply src/components/daterangepicker/daterangepicker.vue]: Great callout -- this made me think of how I could use `jQuery` to change the value. I've deprecated `setDefaultText` as a global in this file and converted it to a method that accesses the `jQuery.val` to update the text

**Comment by SashaBayan** [comment]
Thanks for all the reviews! Great improvements from the MVP till now -- all the input has been much appreciated here
